docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -210,85 +211,94 @@ class HttpServerExampleSpec extends AnyWordSpec with Matchers
 
     // backend entry points
     def myAuthenticator: Authenticator[User] = ???
-    def retrieveOrdersFromDB: Seq[Order] = ???
+    def retrieveOrdersFromDB: Future[Seq[Order]] = ???
     def myDbActor: ActorRef = ???
     def processOrderRequest(id: Int, complete: Order => Unit): Unit = ???
 
-    val route = concat(
-      path("orders") {
-        authenticateBasic(realm = "admin area", myAuthenticator) { user =>
+    lazy val binding = Http().newServerAt("localhost", 8080).bind(topLevelRoute)
+    // ...
+
+    lazy val topLevelRoute: Route =
+      // provide top-level path structure here but delegate functionality to subroutes for readability
+      concat(
+        path("orders")(ordersRoute),
+        // extract URI path element as Int
+        pathPrefix("order" / IntNumber)(orderRoute),
+        pathPrefix("documentation")(documentationRoute),
+        path("oldApi" / Remaining) { pathRest =>
+          redirect("http://oldapi.example.com/" + pathRest, MovedPermanently)
+        }
+      )
+
+    lazy val ordersRoute: Route =
```

Comment:
perhaps noted elsewhere, so sorry if I missed that, but perhaps put add a comment that the different subroute trees can be defined in different classes as well

Reply:
```suggestion
    // For bigger routes, these sub-routes can be moved to separate files
    lazy val ordersRoute: Route =
```